packages/dd-trace/src/exporters/agent/writer.js

```diff
@@ -26,6 +26,15 @@ class Writer {
   _sendPayload (data, count) {
     platform.metrics().increment(`${METRIC_PREFIX}.requests`, true)
 
+    let done = () => {}
+    const thenable = {
+      then (fulfilled) {
+        // We're not ever going to reject, because users of the thenable are
+        // only concerned about completion, rather than success.
+        done = fulfilled
```

Comment:
What if `then` is called after the operation is done?

Reply:
Yeah thenables may not be the right approach.